### PR TITLE
feat: diff request & other correlation assignments

### LINKED ISSUE #410: Differentiate between request and other types of telemetry when assigning correlation
**Is your feature related to a problem? Please describe.**
Currently, we don't differentiate between request and other types of telemetry when we assign the correlation information before we track the telemetry in Application Insights. The leads to a problem of parent linking as the dependencies have the same parent as the request, while the dependencies (and other types) should have as parent the incoming request.

**What solution do you propose?**
We should differentiate between request telemetry and other types of telemetry in the `OperationContextConverter` in the `Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights` project.

When assigning correlation to request telemetry, these properties should be affected:
- `Id` = `Correlation.OperationId` or generated from request tracking options if not defined
- `Context.Operation.Id` = `Correlation.TransactionId`
- `Context.Operation.ParentId` = `Correlation.OperationParentId`

When assigning correlation to other types of telemetry, these properties should be affected:
- `Context.Operation.Id` = `Correaltion.TransactionId`
- `Context.Operation.ParentId` = `Correlation.OperationId`

### PR DESCRIPTION
**Pull request by stijnmoreels**
Differentiate between request and other types of telemetry tracking.

Closes #410

### CODE REVIEW

File: src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/RequestTelemetryConverter.cs [hunk 1]
```diff
@@ -67,12 +76,16 @@ protected override RequestTelemetry CreateTelemetryEntry(LogEvent logEvent, IFor
 
             var requestTelemetry = new RequestTelemetry(sourceName, requestTime, requestDuration, responseStatusCode, isSuccessfulRequest)
             {
-                Id = id,
                 Url = url,
                 Source = source,
                 Context = { Operation = { Name = requestOperationName } }
             };
 
+            if (_options != null)
```
Comment (fgheysels) [inline src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/RequestTelemetryConverter.cs:84]: Can `_options` ever be null ?

File: src/Arcus.Observability.Tests.Integration/Serilog/Sinks/ApplicationInsights/EventTests.cs [hunk 2]
```diff
@@ -161,8 +161,8 @@ public async Task LogEventWithCorrelationInfo_SinksToApplicationInsights_Results
                         Assert.Equal(operationId, actualOperationId);
                         Assert.Equal(transactionId, actualTransactionId);
                         Assert.Equal(operationParentId, actualOperationParentId);
-                        Assert.Equal(operationId, trace.Operation.Id);
-                        Assert.Equal(operationParentId, trace.Operation.ParentId);
+                        Assert.Equal(transactionId, trace.Operation.Id);
+                        Assert.Equal(operationId, trace.Operation.ParentId);
```
Comment (fgheysels) [inline src/Arcus.Observability.Tests.Integration/Serilog/Sinks/ApplicationInsights/EventTests.cs:165]: I might be missing some context, but I think this is strange ?

File: src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/RequestTelemetryConverter.cs (hunk 1 above)
Comment (stijnmoreels) [reply src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/RequestTelemetryConverter.cs:84]: Yes, bc this implementation makes sure we support both the 'old' way of injecting the request-specific options, and the 'new' way of options passed-along via the base class.

File: src/Arcus.Observability.Tests.Integration/Serilog/Sinks/ApplicationInsights/EventTests.cs (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.Observability.Tests.Integration/Serilog/Sinks/ApplicationInsights/EventTests.cs:165]: This is the conclusion from the service-to-service PR we had last week. The non-requests will need some different telemetry correlation to be fully linked to the incoming requests.

Also, Application Insights' `Operation ID` = `Transaction ID` from Arcus.

File: src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/RequestTelemetryConverter.cs (hunk 1 above)
Comment (stijnmoreels) [reply src/Arcus.Observability.Telemetry.Serilog.Sinks.ApplicationInsights/Converters/RequestTelemetryConverter.cs:84]: The ID is now set in the `OperationContextConverter` where the correlation assignment is centralized. There, the operation ID is being used as the ID of the request (like proposed in the service-to-service poc).

File: src/Arcus.Observability.Tests.Integration/Serilog/Sinks/ApplicationInsights/EventTests.cs (hunk 2 above)
Comment (stijnmoreels) [reply src/Arcus.Observability.Tests.Integration/Serilog/Sinks/ApplicationInsights/EventTests.cs:165]: 😄 I can only imagine what that would be like, constantly switching. You're doing a fine job! 👍

**Review by fgheysels — APPROVED** [review]
I think this is ok :)

**Pull request merged**